src/server/logger.js

```diff
@@ -1,27 +1,97 @@
 'use strict';
 
-var levels = ['trace', 'info', 'log', 'debug', 'warn', 'error'],
-    debug = require('debug');
+const chalk = require('chalk');
+
+const LEVELS = ['trace', 'info', 'log', 'debug', 'warn', 'error'],
+    LEVELCOLORS = {
+        'trace': {
+            bg: 'bgBlack',
+            fg: 'white'
+        }, 
+        'info': {
+            bg: 'bgBlack',
+            fg: 'white'
+        },
+        'log': {
+            bg: 'bgBlack',
+            fg: 'white'
+        },
+        'debug': {
+            bg: 'bgBlack',
+            fg: 'white'
+        },
+        'warn':  {
+            bg: 'bgYellow',
+            fg: 'black'
+        },
+        'error': {
+            bg: 'bgRed',
+            fg: 'white'
+        }
+    };
 
 // set which debug levels to send to stderr, the rest will go to stdout
 const STDERR = ['warn', 'error'];
 
 var Logger = function(name) {
     this._name = name;
-    /* eslint-disable no-console*/
-    levels.forEach(lvl => {
-        this[lvl] = debug(`${name}:${lvl}`);
-        if (STDERR.find(item => item === lvl)) {
-            this[lvl].log = console.error.bind(console);
-        } else { // send to stdout
-            this[lvl].log = console.log.bind(console);
-        }
+    LEVELS.forEach(lvl => {
+        this[lvl] = this._log.bind(this, lvl);
     });
+};
+
+Logger.prototype._log = function(level, content) {
+
+    // Allow skips
+    if(Logger.prototype.skips.some((skip) => skip.test(this._name)))
+    {
+        return;
+    }
+
+    // Allow whitelist
+    if(Logger.prototype.names.length > 0){
+        
+        if(!Logger.prototype.names.some((name) => name.test(this._name)))
+        {
+            return;
+        }
+    }
+
+    /* eslint-disable no-console*/
+    // Determine which output to use
+    let logFunc = STDERR.find(lvl => level === lvl)? console.error : console.log;
+    logFunc = logFunc.bind(console);
     /* eslint-enable no-console*/
+
+    // Prevent issues if no color available
+    logFunc(chalk[LEVELCOLORS[level].bg](chalk[LEVELCOLORS[level].fg](`${Date.now()} ${this._name}:${level}`)) + ' ' + content);
```

Comment:
date in a human readable format might be better
then maybe we could shorten it to something like: `month.day.hour.min.sec`

Reply:
Date format is now set by DEBUG_DATE in the env. Do you want the default to be MM.DD.HH.mm.ss?

Reply:
right now it has a space between day and hour I don't and colon for the hour which looks good. but I slightly lean toward not using space in the default format. 
if there is no space then we can kinda assume space to be the column delimiter which separates time, namespaces and the logs. maybe `-` instead of that space?